src/mlir/dialect/VeronaTypes.h

```diff
@@ -41,11 +41,15 @@ namespace mlir::verona
     struct JoinTypeStorage;
     struct IntegerTypeStorage;
     struct FloatTypeStorage;
+    struct DescriptorTypeStorage;
     struct CapabilityTypeStorage;
     struct ClassTypeStorage;
     struct ViewpointTypeStorage;
   }
 
+  /**
+   * Meet types are intersections between types (A & B).
```

Comment:
Is there a reason to invent a new word for them here?

Reply:
@plietar chose the name. I don't mind either way.

Reply:
If we want to choose a new name, it shouldn't be done in this PR anyway, so I'll mark this as resolved.

Reply:
I don't know if David's "new word" was referring to Meet or to Intersection.

I chose meet/join over intersection/union for a few reasons, none of which are particularily strong:
- Union is a C++ keyword
- Meet and Join are short, and of the same length
- Intersection is hard to type
- (Maybe the most valid reason) there are various interpretations of the words "union" and "intersection", used in different languages. There's the C/C++/Rust unions, there's discriminated unions (ie. Haskell like data types), there's the ["implictly projected pairs of values" intersections](0), and there's the Scala-like intersection. None of these have the Pony/Verona semantics.
On the other hand, Meet and Join have, I believe, precise and unambiguous semantics. They are respectively the greatest lower bound and least-upper bound in the subtyping lattice.

[0]: https://dl.acm.org/doi/pdf/10.1145/2951913.2951945

Reply:
I personally like `Meet` and `Join` for the same reasons @plietar explain above.